src/psyclone/tests/lfric_build_test.py

```diff
@@ -0,0 +1,152 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2023, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author:  A. R. Porter, STFC Daresbury Lab
+
+''' This module contains tests for the lfric_build.py file. '''
+
+import os
+import subprocess
+import pytest
+
+from psyclone.tests.lfric_build import LFRicBuild
+from psyclone.tests.utilities import Compile, CompileError
+
+
+@pytest.fixture(name="enable_compilation")
+def protect_infrastructure_path_fixture(monkeypatch):
+    '''A pytest fixture that mocks testing with compilation enabled while
+    ensuring that any pre-existing LFRicBuild._compilation_path is
+    preserved.'''
+    # Pretend that the infrastructure has already been built so that the
+    # LFRicBuild constructor doesn't attempt to trigger it.
+    monkeypatch.setattr(LFRicBuild, "_infrastructure_built", True)
+    # Pretend that compilation testing is enabled.
+    if not Compile.TEST_COMPILE:
+        monkeypatch.setattr(Compile, "TEST_COMPILE", True)
+    else:
```

Comment:
I believe you can just remove the `else`, so that the `monkeypatch` is executed for both cases (originally compile and non-compile). Then this line would be covered, and at least one very quick test worked with that (and it did not work if I removed the monkeypatch, so I think I tested the right thing).

Reply:
You are right! I just removed the `if:...else:` altogether and it all works. Happy days :-)